Sources/PortalSwift/Core/Api/PortalApi.swift

```diff
@@ -0,0 +1,589 @@
+  ///   - data: (Optional) The transacton "data" parameter.
+  ///   - maxFeePerGas: (Optional) The transacton "maxFeePerGas" parameter.
+  ///   - maxPriorityFeePerGas: (Optional) The transacton "maxPriorityFeePerGas" parameter.
+  ///   - gas: (Optional) The transacton "gas" parameter.
+  ///   - gasPrice: (Optional) The transacton "gasPrice" parameter.
+  ///   - completion: The callback that contains transaction simulation response.
+  /// - Returns: Void.
+  public func simulateTransaction(_ transaction: Any, withChainId: String) async throws -> SimulatedTransaction {
+    guard let chainId = withChainId.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else {
+      throw PortalApiError.unableToEncodeData
+    }
+    if let url = URL(string: "\(baseUrl)/api/v3/clients/me/simulate-transaction?chainId=\(chainId)") {
+      do {
+        let transformedTransaction = try AnyEncodable(transaction)
+        let data = try await post(url, withBearerToken: self.apiKey, andPayload: transformedTransaction)
+        let simulatedTransaction = try decoder.decode(SimulatedTransaction.self, from: data)
+
+        return simulatedTransaction
+      } catch {
+        self.logger.error("PortalApi.simulateTransaction() - Unable to simulate transaction: \(error.localizedDescription)")
+        throw error
+      }
+    }
+
+    self.logger.error("PortalApi.simulateTransaction() - Unable to build request URL.")
+    throw URLError(.badURL)
+  }
+
+  func track(_ event: String, withProperties: [String: AnyEncodable]) async throws -> MetricsResponse {
+    if let url = URL(string: "\(baseUrl)/api/v1/analytics/track") {
+      let payload = MetricsTrackRequest(
+        event: event,
+        properties: withProperties
+      )
+      let data = try await post(url, withBearerToken: self.apiKey, andPayload: payload)
+      let response = try decoder.decode(MetricsResponse.self, from: data)
+
+      return response
+    }
+
+    throw URLError(.badURL)
+  }
+
+  public func updateShareStatus(
+    _ type: PortalSharePairType,
+    status: SharePairUpdateStatus,
+    sharePairIds: [String]
+  ) async throws {
+    if let url = URL(string: "\(baseUrl)/api/v3/clients/me/\(type.rawValue)-share-pairs/") {
+      do {
+        let payload = ShareStatusUpdateRequest(
+          backupSharePairIds: type == .backup ? sharePairIds : nil,
+          signingSharePairIds: type == .signing ? sharePairIds : nil,
+          status: status
+        )
+
+        _ = try await self.patch(url, withBearerToken: self.apiKey, andPayload: payload)
+
+        return
+      } catch {
+        self.logger.error("PortalApi.updateShareStatus() - Unable to update share status: \(error.localizedDescription)")
+        throw error
+      }
+    }
+
+    self.logger.error("PortalApi.updateShareStatus() - Unable to build request URL.")
+    throw URLError(.badURL)
+  }
+
+  /*******************************************
+   * Private functions
```

Comment:
Simple wrapper of the `requests` property to simplify the code in the above `public` functions.